Add Basic Auth Support to Signal Messenger

CodeShellDev opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This PR adds Basic Auth support for the Signal Messenger Integration, while this is already supported by `pysignalclirestapi`, there was no configuration for it in the homeassistant config.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: None
- This PR is related to issue: None
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/41589
- Link to developer documentation pull request: None
- Link to frontend pull request: None

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.

  AI tools are welcome, but contributors are responsible for *fully*
  understanding the code before submitting a PR.
-->

- [x] I understand the code I am submitting and can explain how it works.
- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.
- [ ] Any generated code has been carefully reviewed for correctness and compliance with project standards.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

CodeShellDev committed:
add basic auth support to signal-messenger

CodeShellDev committed:
use param instead of private member

CodeShellDev committed:
Merge branch 'dev' into signal-messenger-basic-auth

CodeShellDev committed:
Merge branch 'dev' into signal-messenger-basic-auth

pantherale0 commented on homeassistant/components/signal_messenger/notify.py:
```diff
@@ -25,6 +29,9 @@
 CONF_MAX_ALLOWED_DOWNLOAD_SIZE_BYTES = 52428800
+CONF_BASIC_AUTH = "auth"
+CONF_BASIC_AUTH_USER = "user"
+CONF_BASIC_AUTH_PW = "password"
```
I think you can use Home Assistant's default consts for these:

```
from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
```

CodeShellDev replied:
Are the consts also „user“ and „password“ (need to know for docs)

pantherale0 replied:
`username` and `password`:

https://github.com/home-assistant/core/blob/92fbf468f2ef477bd88448bc6f52dca3f4e64d40/homeassistant/const.py#L191
https://github.com/home-assistant/core/blob/92fbf468f2ef477bd88448bc6f52dca3f4e64d40/homeassistant/const.py#L253

CodeShellDev replied:
Ok, will update in docs

CodeShellDev committed:
Implementes hass consts for user and password (proposed by reviewer)

CodeShellDev committed:
Merge branch 'dev' into signal-messenger-basic-auth

CodeShellDev committed:
Merge branch 'dev' into signal-messenger-basic-auth

CodeShellDev committed:
Merge branch 'dev' into signal-messenger-basic-auth

CodeShellDev committed:
Merge branch 'dev' into signal-messenger-basic-auth

CodeShellDev committed:
Merge branch 'dev' into signal-messenger-basic-auth

CodeShellDev committed:
Change password input to use TextSelector

CodeShellDev committed:
Fix indentation in signal_messenger notify.py

CodeShellDev committed:
Removed TextSelector

Removed `TextSelectorType.PASSWORD`,
due to believe, that it is only needed and used for UI-based configurations. Which Signal Messenger doesn‘t support because of it being a Legacy Integration.

CodeShellDev committed:
Merge branch 'dev' into signal-messenger-basic-auth

CodeShellDev committed:
Merge branch 'dev' into signal-messenger-basic-auth

joostlek reviewed (changes requested):
We no longer allow integrations to add or change a platform YAML configuration.

More information on this can be found in Architecture Decision Record:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>

Please note that this integration connects to a device or service and another Architecture Decision Record applies that disallows the use of YAML configuration in favor of a configuration flow via the UI:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

We (and our community!) would really appreciate it if you could start on the refactoring of this integration in a new PR. Please let us know if you need help with this :)

Thanks already! :+1:
